### PR TITLE
Add account verification module and role

### PR DESCRIPTION
**Pull request by patrickeasters**
This adds the first attempt at a cloud_provider_module. I'm still cleaning up some error handling, so marking this still as a draft.

### CODE REVIEW

File: plugins/modules/cloud_provider_account.py [hunk 1]
```diff
@@ -0,0 +1,201 @@
...
+    description:
+      - Short Name of the cloud provider
+    choices: [ AWS, MSAZ, GCP ]
+    required: true
+    type: str
+  id:
+    description:
+      - Account ID of the cloud provider account
+    required: true
+    type: strss
```
Comment (hseljenes) [inline plugins/modules/cloud_provider_account.py]: Typo?

File: plugins/modules/cloud_provider_account.py [hunk 2]
```diff
@@ -0,0 +1,201 @@
...
+    for p in providers['body']:
+        if p['shortName'] == module.params['provider']:
+            for a in p['accounts']:
+                if a['id'] == module.params['id']:
+                    account = a
+                    break
+            break
+    
+    # Delete account if needed
+    if account and module.params['provider'] == 'present':
```
Comment (hseljenes) [inline plugins/modules/cloud_provider_account.py]: How does this work? `provider` is one of `['AWS', 'MSAZ', 'GCP']`, right?

File: plugins/modules/cloud_provider_account.py [hunk 3]
```diff
@@ -0,0 +1,201 @@
...
+
+    module.exit_json(**result)
+
+
+def main():
+    run_module()
+
+
+if __name__ == '__main__':
+    main()
```
Comment (hseljenes) [inline plugins/modules/cloud_provider_account.py]: Do we care about newlines at end of files? I have my editor set to add them by default but I've never really encountered a tool that fails without it present.

File: plugins/modules/cloud_provider_account.py [hunk 4]
```diff
@@ -0,0 +1,201 @@
...
+  refresh_token_client:
+    description:
+      - Client ID used to generate the refresh token
+      - Typically 'rhsm-api' for tokens generated at access.redhat.com and 'cloud-services' for cloud.redhat.com
+    default: rhsm-api
+    type: str
+  provider:
+    description:
+      - Short Name of the cloud provider
+    choices: [ AWS, MSAZ, GCP ]
```
Comment (hseljenes) [inline plugins/modules/cloud_provider_account.py:35]: Should we drop `GCP` here given it's not supported yet?

**Review by hseljenes — COMMENTED** [review]
Adding some feedback while I go try this out.

File: plugins/modules/cloud_provider_account.py (hunk 3 above)
Comment (patrickeasters) [reply plugins/modules/cloud_provider_account.py]: Probably should for consistency's sake.

File: plugins/modules/cloud_provider_account.py (hunk 2 above)
Comment (patrickeasters) [reply plugins/modules/cloud_provider_account.py]: Yea, that's what I get for getting distracted while copy/pasting. 🙃

File: plugins/modules/cloud_provider_account.py (hunk 4 above)
Comment (hseljenes) [reply plugins/modules/cloud_provider_account.py:35]: "it's still useful in the module since you can create a provider without verifying it"

**Review by hseljenes — APPROVED** [review]
lgtm

**Pull request merged**